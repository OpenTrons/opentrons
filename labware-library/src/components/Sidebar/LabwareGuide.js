// @flow
// labware filters
import * as React from 'react'

import { Icon } from '@opentrons/components'
import styles from './styles.css'

// TODO(mc, 2019-03-14): i18n
const EN_LABWARE_GUIDE = 'Labware Guide'
const EN_HOW_TO_CHOOSE_LABWARE = 'How to choose labware'
const EN_MAKING_CUSTOM_DEFINITIONS = 'Making custom definitions'

const LINKS = [
  { label: EN_HOW_TO_CHOOSE_LABWARE, href: '#' },
  { label: EN_MAKING_CUSTOM_DEFINITIONS, href: '#' },
]

export default function LabwareGuide() {
  return (
    <section className={styles.labware_guide}>
      <div className={styles.labware_guide_container}>
        <h2 className={styles.labware_guide_title}>
          <Icon
            className={styles.labware_guide_icon}
            name="book-open-page-variant"
          />
          {EN_LABWARE_GUIDE}
        </h2>
        <ul className={styles.labware_guide_list}>
          {LINKS.map((a, i) => (
            <li key={i}>
              <a
                className={styles.labware_guide_link}
                href={a.href}
                target="_blank"
                rel="noopener noreferrer"
              >
                {a.label}
              </a>
            </li>
          ))}
        </ul>
      </div>
    </section>
  )
}
