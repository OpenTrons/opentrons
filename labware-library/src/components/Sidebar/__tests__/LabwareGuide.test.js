// @flow
// LabwareGuide component tests
import * as React from 'react'
import Renderer from 'react-test-renderer'

import LabwareGuide from '../LabwareGuide'

describe('LabwareGuide', () => {
  test('component renders', () => {
    const tree = Renderer.create(<LabwareGuide />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
