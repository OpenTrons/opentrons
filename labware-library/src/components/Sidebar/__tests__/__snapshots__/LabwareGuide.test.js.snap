// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabwareGuide component renders 1`] = `
<section
  className="labware_guide"
>
  <div
    className="labware_guide_container"
  >
    <h2
      className="labware_guide_title"
    >
      <Icon
        className="labware_guide_icon"
        name="book-open-page-variant"
      />
      Labware Guide
    </h2>
    <ul
      className="labware_guide_list"
    >
      <li
        key="0"
      >
        <a
          className="labware_guide_link"
          href="https://support.opentrons.com/en/articles/3136501-what-is-a-labware-definition"
          rel="noopener noreferrer"
          target="_blank"
        >
          What is a labware definition?
        </a>
      </li>
      <li
        key="1"
      >
        <a
          className="labware_guide_link"
          href="https://support.opentrons.com/en/articles/3136507-using-the-labware-library"
          rel="noopener noreferrer"
          target="_blank"
        >
          Using the Labware Library
        </a>
      </li>
      <li
        key="2"
      >
        <a
          className="labware_guide_link"
          href="https://support.opentrons.com/en/articles/3136504-creating-custom-labware-definitions"
          rel="noopener noreferrer"
          target="_blank"
        >
          Creating custom labware definitions
        </a>
      </li>
      <li>
        <Link
          className="labware_guide_link"
          to="/create"
        >
          Custom Labware Creator
        </Link>
      </li>
    </ul>
  </div>
</section>
`;
